File: SuperHero/app/src/main/java/pe/isil/superhero/Hero.java
package pe.isil.superhero;

import com.google.gson.annotations.SerializedName;

public class Hero {

    private String name;

    private Biography biography;

    @SerializedName("image")
    private HeroImage heroImage;

    public String getName() {
        return name;
    }

    public String getFullName() {
        return biography.getFullName();
    }

    public String getUrl() {
        return heroImage.getUrl();
    }


    private class Biography {

        @SerializedName("full-name")
        private String fullName;

        public String getFullName() {
            return fullName;
        }
    }

    private class HeroImage {

        private String url;

        public String getUrl() {
            return url;
        }
    }
}

File: SuperHeroCompose/app/src/main/java/pe/isil/superherocompose/data/model/Hero.kt
package pe.isil.superherocompose.data.model

import com.google.gson.annotations.SerializedName

data class HeroResponse(
    @SerializedName("results")
    val heroes: ArrayList<Hero>
)

data class Hero (
    val name: String,
    val biography: Biography,
    val image: HeroImage
)

data class Biography (
    @SerializedName("full-name")
    val fullName: String
)

data class HeroImage (
    val url: String
)
